apis/python/requirements_dev.txt

```diff
@@ -3,5 +3,7 @@ ruff
 pytest
 pytest-cov
 sparse
-typeguard==4.2.1
+# Python 3.12 support requires https://github.com/agronholm/typeguard/pull/490.
+# Use Typeguard @ HEAD until that PR is included in a release.
+typeguard @ git+https://github.com/agronholm/typeguard
```

Comment:
This would be the place for a TODO / issue link related to upgrading Typeguard. Do you want me to change this comment?

Reply:
Filed #3216 and referenced it here

Reply:
Thanks @ryan-williams !